nestkernel/source_table.cpp

```diff
@@ -408,3 +448,103 @@ nest::SourceTable::get_next_target_data( const thread tid,
     return true; // found a valid entry
   }
 }
+
+void
+nest::SourceTable::resize_compressible_sources()
+{
+  for ( thread tid = 0; tid < static_cast< thread >( compressible_sources_.size() ); ++tid )
+  {
+    compressible_sources_[ tid ].clear();
+    compressible_sources_[ tid ].resize(
+      kernel().model_manager.get_num_synapse_prototypes(), std::map< index, SpikeData >() );
+  }
+}
+
+void
+nest::SourceTable::collect_compressible_sources( const thread tid )
+{
+  for ( synindex syn_id = 0; syn_id < sources_[ tid ].size(); ++syn_id )
+  {
+    index lcid = 0;
+    while ( lcid < sources_[ tid ][ syn_id ].size() )
```

Comment:
To make this code more readable, I'd introduce
```c++
auto &syn_sources = sources_[ tid ][ syn_id ];
```
and then use that in what follows.

Reply:
ok